File: 12.c
#include <stdio.h>

int yakusuu(int x)
{
	int i;
	int counter = 0;
	for (i = 1; i <= x; i++)
		if (x % i == 0)
			counter++;

	return counter;
}

void main()
{
	int bak, num, i, j, tmp;
	int counter=2;

	i = 1;
	tmp = 0;

//	while (i <= 10) {
	while (1) {
/*
		for (tmp = 0, j = 1; j <= i; j++)
			tmp += j;
*/
//printf("%d\n", tmp);
		tmp += i;
		i++;
		if (yakusuu(tmp) >= 501)
			break;
	}
printf("%d\n", tmp);
}

File: 12.rb
#!/usr/bin/ruby -Ku

def yakusuu(y)
  total = 1
  i = 2
  x = Hash.new(0)

  while i <= y do 
    if (y % i) == 0
      x[i] += 1
      y /= i
    else
      i += 1
    end
  end

  x.each do |key, value|
    total *= (value + 1)
#    total = total * (value + 1)
  end

  return total
end

num = 1
tmp = 1

while 1 do
  if yakusuu(tmp) >= 501
    puts tmp
    break
  end

  num += 1
  tmp += num
end

=begin

def yakusuu(num)
  total = 1
  i = 1
  x = Hash.new(0)

  while i <= num
    if (i % num) == 0
      x[i] += 1
    else
      i += 1
    end
  end

  x.each do |xx|
    total *= (xx + 1)
  end

  return total + 2
end
=end
